domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/validation/casefield/CaseFieldEntityComplexFieldACLValidatorImplTest.java

```diff
@@ -0,0 +1,274 @@
+        complexACL1.setCreate(true);
+        complexACL1.setUserRole(userRole);
+        fieldACL.setCreate(true);
+        fieldACL.setUserRole(userRole);
+        caseField.addComplexFieldACL(complexACL1);
+        caseField.addCaseFieldACL(fieldACL);
+
+        final ValidationResult result = validator.validate(caseField, caseFieldEntityValidationContext);
+
+        final List<ValidationError> validationErrors = result.getValidationErrors();
+        assertAll(
+            () -> assertThat(validationErrors, not(empty())),
+            () -> assertThat(validationErrors.get(0).getDefaultMessage(), is("Parent list element code 'Class' is missing for list element code 'Class.ClassMembers'"))
+        );
+    }
+
+    @Test
+    @DisplayName("should fail when complex child element has more access than parent")
+    public void shouldFailWhenComplexChildHasMoreAccessThanCaseFieldParent() {
+        complexACL0.setUserRole(userRole);
+        complexACL0.setCreate(true);
+        fieldACL.setCreate(false);
+        fieldACL.setUserRole(userRole);
+        caseField.addComplexFieldACLEntities(asList(complexACL0));
+        caseField.addCaseFieldACL(fieldACL);
+
+        final ValidationResult result = validator.validate(caseField, caseFieldEntityValidationContext);
+
+        final List<ValidationError> validationErrors = result.getValidationErrors();
+        assertAll(
+            () -> assertThat(validationErrors, not(empty())),
+            () -> assertThat(validationErrors.get(0).getDefaultMessage(), is("List element code 'Class' has more"
+                + " access than case field 'case_field'"))
+        );
+    }
+
+    @Test
+    @DisplayName("should fail when a complex child has more access than its complex parent")
+    public void shouldFailWhenComplexChildHasMoreAccessThanComplexParent() {
+        complexACL0.setUserRole(userRole);
+        complexACL0.setCreate(false);
+        complexACL1.setUserRole(userRole);
+        complexACL1.setCreate(false);
+        complexACL2.setUserRole(userRole);
+        complexACL2.setCreate(true);
+        caseField.addComplexFieldACLEntities(asList(complexACL0, complexACL1, complexACL2));
+        fieldACL.setUserRole(userRole);
+        fieldACL.setCreate(true);
+        caseField.addCaseFieldACL(fieldACL);
+
+        final ValidationResult result = validator.validate(caseField, caseFieldEntityValidationContext);
+
+        assertAll(
+            () -> assertThat(result.getValidationErrors(), not(empty())),
+            () -> assertThat(result.getValidationErrors().get(0).getDefaultMessage(), is("List element code "
+                + "'Class.ClassMembers.Children' has more access than its parent 'Class'"))
+        );
+    }
+
+    @Test
+    @DisplayName("should fail when a complex child has more access than its second level complex parent")
+    public void shouldFailWhenComplexChildHasMoreAccessThanSecondLevelComplexParent() {
```

Comment:
Are we sure this is failing due to access or missing parent - `complexACL0`?

Reply:
Or in other words - don't we need the root of the dot notation path to exist in on the field?

Reply:
yes, this is how we decided with @amansinghb

Reply:
Actually it failed for both, so altered setup to make it clearer and tightened validation 👍